### DISCUSSION TITLE
Configuring the ssh kitten, problems with cwd and base64

### DISCUSSION THREAD

**Discussion by gzagatti** [kovidgoyal/kitty#5084] [Q&A]
This is my first time using ssh with `kitty`. As recommended in the manual, I have started by using the kitten ssh. Most of the stuff is working for me, but I am facing a few hiccups which I cannot solve.

When I log in to the remote server, I get a function printed to my terminal called `die`. It is printed in a somewhat garbled fashion. It seems like it is complaining that command `base64` is missing from the remote server. Despite the message, I am able to use the remote server without a problem. Is there something I could install in the remote to support `base64`?

I tried to use the option `--kitten interpreter=python3` when ssh'ing which solves the problem with the garbled message. However, as my username in the remote host is different, whenever I try to open a new window, I get an error message saying that `/home/<host-username>` does not exist and the window gets closed.

In the `ssh.conf`, if I set `cwd "$HOME"`. I now get an error saying that `/home/<remote-username>` does not exist and the connection does not even start. This is strange because the path does indeed exist in the remote. How could I set `cwd` in `ssh.conf`?

**Comment by kovidgoyal** [comment]
1) Install base64 on the server.
2) That seems like a bug. Post the full errror message you get and are you trying to open a new window using the same cwd?

**Comment by ShoMinamimoto** [comment]
I have the same issue with base64, although it only happens with charm apps like [Soft Serve](https://github.com/charmbracelet/soft-serve) and [Wishlist](https://github.com/charmbracelet/wishlist), even when I run them locally, where base64 is definitely installed.

The server log for connection attempts via the kitten look like this:
```
{ Distributed under terms of the GPLv3 license.dgoyal.net>xec sh -c eval "$(echo "$0" | tr \\\v\\\f\\\r\\\b \\\047\\\134\\\n\\\041)" /bin/sh
  unalias command; 
die() { printf " ] && command rm -rf "$tdir"ho" 2> /dev/null < /dev/tty
                033[31m%s
                         033[m
                              n
    base64_encode() { command base64 | command tr -d turn 1; fi 1; fistrap_exit; exit 1; }

                                                     n

    base64_encode() { command b64encode - | command sed  }
                                                        1d;$d
                                                              | command tr -d 

                                                                              n

    pybase64() { command "$python" -c "import sys, base64; getattr(sys.stdout, r; }
                                                                               buffer
                                                                                     , sys.stdout).write(base64.standard_b64$1(getattr(sys.stdin, 
                                                                                                                                                  buffer
    base64_encode() { command "$perl" -MMIME::Base64 -0777 -ne                                                                                          , sys.stdin).read()))"; }
                                                               print encode_base64($_)
    base64_decode() { command "$perl" -MMIME::Base64 -ne                              ; }
                                                         print decode_base64($_)
dcs_to_kitty() { printf "e not present on remote host, ssh kitten cannot function."
                         033P@kitty-$1|%s
                                         033
] xterm-kitty 213 57assed to SSH execute it here"thenoot" ""ad SSH data from tty" /dev/nulle in the future
```

For comparison without the kitten:
```
2022/05/24 11:21:14 rainer connect 127.0.0.1:48990 true [] xterm-kitty 213 57
```

**Reply by kovidgoyal** [reply]
This will be because whatever environment the charm apps use is not a proper POSIX sh environment, so things like the ssh kitten are unlikely to work with it.

**Comment by enrichilversum** [comment]
Hi @nenahp 
Try : 
`kitten @ launch blablabla --kitten cwd=/foo/bar`

The --kitten argument allows passing ssh arguments as values. I just had a similar issue (and our server does have a proper POSIX sh, we are a large academic site)

(I know this is old, but somebody else could be searching, just like I did)
